pkg/clusterinfo/clusterinfo_test.go

```diff
@@ -85,7 +89,9 @@ func newMockConsumer(ctx context.Context, name string, t *testing.T) *mockConsum
 		for {
 			select {
 			case d := <-mc.ch:
+				mc.mu.Lock()
 				mc.data = d
+				mc.mu.Unlock()
 				t.Logf("consumer %s received data from channel: %+v\n", mc, mc.data)
```

Comment:
there should still be a lock here. I would not take a read lock but extend the write log.

Reply:
Thank you for the feedback! Nice catch. This should now be resolved.